README.md
# NgGraphics &middot; [![npm](https://img.shields.io/npm/dw/ng-graphics?style=flat-square)](https://www.npmjs.com/package/ng-graphics)

A d3-driven charting library for Angular applications 📊
contributor friendly and idiomatic 🤗

## Installing / Getting started

First Install this dependency:

```shell
npm i ng-graphics
```

Then put the following line into your **app.module.ts** file and add that variable to your **imports** array:

```typescript
import { NgChartsModule } from "ng-graphics";
```

Add the following to lines to your **angular.json**:

```text
"node_modules/d3/dist/d3.min.js",
"node_modules/jquery/dist/jquery.min.js",
"node_modules/popper.js/dist/umd/popper.min.js",
"node_modules/tippy.js/dist/tippy-bundle.iife.min.js"
```

If for instance you want a bar-chart, add this to your html:

```html
<div style="height: 500px; width: 500px">
  <lib-bar-chart
    [data]="[
      {
        key: 'A',
        value: 30
      },
      {
        key: 'B',
        value: 50
      }
    ]"
  ></lib-bar-chart>
</div>
```

It will inherit the height and the width of its parent element 🌳

## Developing

### Building

You can build the library by running the following line:

```shell
npm run build:ngCharts
```

and verify that it's working by using the components inside of `src`, run it with:

```shell
npm start
```


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.3.0] - 2019-10-19

### Added

- Add tooltips by Tippy.js
- Add tooltips for Bar-Chart
- Add tooltips for Line-Chart
- Add tooltips for Heatmap
- Add tooltips for Pie-Chart
- Add tooltips for Histogram
- Add tooltips for Scatter-Chart

### Changed

- Change data-keys of Pie-Chart
- Change data-keys of Line-Chart

## [0.2.3] - 2019-10-17

### Added

- Add property `title` to Heatmap
- Add property `title` to Histogram
- Add property `title` to Line-Chart
- Add property `title` to Scatter-Chart
- Add property `title` to Pie-Chart
- Add properties `xlabel` and `ylabel` to Line-Chart
- Add properties `xlabel` and `ylabel` to Histogram
- Add properties `xlabel` and `ylabel` to Scatter-Chart

## [0.2.2] - 2019-10-06

### Added

- Add property `title` to Bar-Chart
- Add properties `xlabel` and `ylabel` to Bar-Chart

## [0.2.0] - 2019-10-02

### Changed

- Rename ng-charts to ng-graphics

## [0.1.3] - 2019-09-30

### Added

- New component for plotting a Line-Chart
- New component for plotting a Heatmap
- New component for plotting a Pie-Chart

### Changed

- Change opacity on mouse-hover events

## [0.1.2] - 2019-09-26

### Added

- New component for plotting a Scatter-Chart
- New component for plotting a Histogram
- Add live-plots for Scatter-Chart and Histogram to the homepage
- Add selector and data-example for each component on the homepage


src/app/app.module.ts
import { NgModule } from '@angular/core'
import { BrowserModule } from '@angular/platform-browser'
import { RouterModule, Routes } from '@angular/router'
import { BrowserAnimationsModule } from '@angular/platform-browser/animations'

import { AppComponent } from './app.component'
import { NgChartsModule } from 'ng-charts'
import material from './app.material'

import { HighlightModule } from 'ngx-highlightjs'
import json from 'highlight.js/lib/languages/typescript'

import { MyNavComponent } from './components/my-nav/my-nav.component'
import { ComponentsComponent } from './components/components/components.component'
import { ComponentComponent } from './components/component/component.component'

const appRoutes: Routes = [
  {
    path: 'components',
    component: ComponentsComponent
  },
  {
    path: 'components/scatterchart',
    component: ComponentComponent,
    data: {
      name: 'Scatterchart',
      selector: 'lib-scatter-chart',
      data: Array.from({length: 3}, () => ({key: Math.random(), value: Math.pow(Math.random(), 2)})),
      properties: [
        {name: 'title', type: 'String', desc: 'Title of the plot'},
        {name: 'xlabel', type: 'String', desc: 'Label which should be shown along the x-achses'},
        {name: 'ylabel', type: 'String', desc: 'Label which should be shown along the y-achses'}
      ]
    }
  },
  {
    path: 'components/histogram',
    component: ComponentComponent,
    data: {
      name: 'Histogram',
      selector: 'lib-histogram',
      data: Array.from({length: 30}, () => Math.floor(Math.random() * 101)),
      properties: [
        {name: 'title', type: 'String', desc: 'Title of the plot'},
        {name: 'xlabel', type: 'String', desc: 'Label which should be shown along the x-achses'},
        {name: 'ylabel', type: 'String', desc: 'Label which should be shown along the y-achses'}
      ]
    },
  },
  {
    path: 'components/piechart',
    component: ComponentComponent,
    data: {
      name: 'Pie-Chart',
      selector: 'lib-pie-chart',
      data: [
        {key: 'bananas', value: 20},
        {key: 'apples', value: 45},
        {key: 'carrots', value: 35}
      ],
      properties: [
        {name: 'title', type: 'String', desc: 'Title of the plot'},
      ]
    }
  },
  {
    path: 'components/heatmap',
    component: ComponentComponent,
    data: {
      name: 'Heatmap',
      selector: 'lib-heatmap',
      data: [
        { group: 'A', key: 'v1', value: 30 },
        { group: 'A', key: 'v2', value: 95 },
        { group: 'B', key: 'v1', value: 37 },
        { group: 'B', key: 'v2', value: 50 }
      ],
      properties: [
        {name: 'title', type: 'String', desc: 'Title of the plot'},
      ]
    }
  },
  {
    path: 'components/linechart',
    component: ComponentComponent,
    data: {
      name: 'Line-Chart',
      selector: 'lib-line-chart',
      data: Array.from({length: 5}, (x, i) => ({key: i, value: Math.random()})),
      properties: [
        {name: 'title', type: 'String', desc: 'Title of the plot'},
        {name: 'xlabel', type: 'String', desc: 'Label which should be shown along the x-achses'},
        {name: 'ylabel', type: 'String', desc: 'Label which should be shown along the y-achses'}
      ]
    }
  },
  {
    path: 'components/barchart',
    component: ComponentComponent,
    data: {
      name: 'Bar-Chart',
      selector: 'lib-bar-chart',
      data: [
        { key: 'A', value: 30 },
        { key: 'B', value: 50 }
      ],
      properties: [
        {name: 'title', type: 'String', desc: 'Title of the plot'},
        {name: 'xlabel', type: 'String', desc: 'Label which should be shown along the x-achses'},
        {name: 'ylabel', type: 'String', desc: 'Label which should be shown along the y-achses'}
      ]
    }
  },
  {
    path: '',
    redirectTo: '/components',
    pathMatch: 'full'
  },
  {
    path: '**',
    component: ComponentsComponent
  }
]

export function hljsLanguages() {
  return [
    {name: 'json', func: json}
  ]
}

@NgModule({
  declarations: [
    AppComponent,
    MyNavComponent,
    ComponentsComponent,
    ComponentComponent
  ],
  imports: [
    BrowserModule,
    NgChartsModule,
    BrowserAnimationsModule,
    ...material,
    RouterModule.forRoot(appRoutes),
    HighlightModule.forRoot({
      languages: hljsLanguages
    })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
